src/main/java/io/anserini/util/DumpIndex.java

```diff
@@ -0,0 +1,221 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.anserini.index.IndexWebCollection;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.analysis.en.EnglishAnalyzer;
+import org.apache.lucene.analysis.util.CharArraySet;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.*;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.store.FSDirectory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class DumpIndex {
+  private static final Logger LOG = LogManager.getLogger(DumpIndex.class);
+
+  class RawDocNotStoredException extends Exception {
+    public RawDocNotStoredException(String message) {
+      super(message);
+    }
+  }
+  class DocVectorNotStoredException extends Exception {
+    public DocVectorNotStoredException(String message) {
+      super(message);
+    }
+  }
+
+  String print_repository_stats( DirectoryReader reader ) throws IOException {
+    int docCount = reader.numDocs();
+    int contentsCount = reader.getDocCount(IndexWebCollection.FIELD_BODY);
+    long termCount = reader.getSumTotalTermFreq(IndexWebCollection.FIELD_BODY);
+
+    StringBuilder sb = new StringBuilder();
+    sb.append("Repository statistics:\n");
+    sb.append("documents:\t" + docCount + "\n");
+    sb.append("contentsCount(doc with contents):\t" + contentsCount + "\n");
+    //sb.append("unique terms:\t" + uniqueTermCount);
+    sb.append("total terms:\t" + termCount + "\n");
+    sb.append("stored fields:\t\t");
+
+    Document d = reader.document(1);
+    List<IndexableField> fields = d.getFields();
+    for (IndexableField f : fields) {
+      sb.append(f.name()+" ");
+    }
+    sb.append("\n");
+    return sb.toString();
+  }
+
+  String print_term_counts( DirectoryReader reader, String termStr )
+          throws IOException, ParseException {
+    StringBuilder sb = new StringBuilder();
+    EnglishAnalyzer ea = new EnglishAnalyzer(CharArraySet.EMPTY_SET);
+    QueryParser qp = new QueryParser(IndexWebCollection.FIELD_BODY, ea);
+    TermQuery q = (TermQuery)qp.parse(termStr);
+    long termFreq = reader.totalTermFreq(q.getTerm());
+    long docCount = reader.docFreq(q.getTerm());
+    sb.append(termStr+" ")
+      .append(q.toString(IndexWebCollection.FIELD_BODY)+" ")
+      .append(termFreq+" ")
+      .append(docCount+" ")
+      .append("\n");
+    return sb.toString();
+  }
+
+  /*
+  * print the internal id
+  */
+  String print_document_id( DirectoryReader reader, ArrayList<String> externalIDs ) throws IOException {
+    StringBuilder sb = new StringBuilder();
+    HashMap<String, Integer> mapping = new HashMap<>();
```

Comment:
Just `Map<String, Integer> mapping`: no need to specify implementation in declaration.